aws-datastore/src/test/java/com/amplifyframework/datastore/DataStoreConfigurationTest.java

```diff
@@ -91,23 +96,27 @@ public void testDefaultOverriddenFromConfiguration() throws JSONException, DataS
      * When building a configuration from both a config file and a configuration object,
      * default values should be overridden, and the provided ones shall be used, instead.
      * @throws JSONException While arranging config file JSON
-     * @throws DataStoreException While building DataStoreConfiguration instances via build()
+     * @throws AmplifyException While building DataStoreConfiguration instances via build(), or when deriving a
+     * {@link ModelSchema} from a {@link Model} class.
      */
     @Test
-    public void testDefaultOverriddenFromConfigurationAndObject() throws JSONException, DataStoreException {
+    public void testDefaultOverriddenFromConfigurationAndObject()
+            throws JSONException, AmplifyException {
         long expectedSyncIntervalMinutes = 6L;
         Long expectedSyncIntervalMs = TimeUnit.MINUTES.toMillis(expectedSyncIntervalMinutes);
         Integer expectedSyncMaxRecords = 3;
         DummyConflictHandler dummyConflictHandler = new DummyConflictHandler();
         DataStoreErrorHandler errorHandler = DefaultDataStoreErrorHandler.instance();
 
-        DataStoreSyncExpression syncExpression = () -> BlogOwner.ID.beginsWith(RandomString.string());
+        DataStoreSyncExpression ownerSyncExpression = () -> BlogOwner.ID.beginsWith(RandomString.string());
+        DataStoreSyncExpression postSyncExpression = () -> Post.ID.beginsWith(RandomString.string());
         DataStoreConfiguration configObject = DataStoreConfiguration
             .builder()
             .syncMaxRecords(expectedSyncMaxRecords)
             .conflictHandler(dummyConflictHandler)
             .errorHandler(errorHandler)
-            .syncExpression(BlogOwner.class, syncExpression)
+            .syncExpression(BlogOwner.class, ownerSyncExpression)
+            .syncExpression("Post", postSyncExpression)
```

Comment:
what happens if the entered string is not a model? where does that fail? is it ignored in runtime?

Reply:
There would be no failure - it would just be ignored.